examples_tests/13.BRDF_Explorer/BRDFExplorerApp.cpp

```diff
@@ -548,21 +548,22 @@ class CDerivativeMapManager
     using Key_t = std::pair<irr::video::IVirtualTexture*, float>;
 
     core::map<Key_t, video::IVirtualTexture*> DerivMaps;
-    IrrlichtDevice* Device;
+    irr::video::CDerivativeMapCreator* DerivMapCreator;
 
 public:
-    CDerivativeMapManager(IrrlichtDevice* _device) : Device{_device} {}
+    CDerivativeMapManager(IrrlichtDevice* _device) : DerivMapCreator(new irr::video::CDerivativeMapCreator(_device->getVideoDriver())) {}
     ~CDerivativeMapManager() {
         for (auto& t : DerivMaps)
             t.second->drop();
+        DerivMapCreator->drop();
```

Comment:
this sort of thing makes  me think we should create a default one in the `IVideoDriver` and just do a getter so there are not 10s of the same thing instantiated.